core/crypto/kem/adapter/kem.go

```diff
@@ -0,0 +1,267 @@
+	_, err = rand.Reader.Read(seed)
+	if err != nil {
+		return
+	}
+	return a.EncapsulateDeterministically(pk, seed)
+}
+
+func hashSharedSecretWithPublicKeys(publicKeySize int, ss []byte, pubkey1 []byte, pubkey2 []byte) []byte {
+	var h blake2b.XOF
+	var err error
+	if len(ss) != 32 {
+		sum := blake2b.Sum256(ss)
+		h, err = blake2b.NewXOF(uint32(publicKeySize), sum[:])
+	} else {
+		h, err = blake2b.NewXOF(uint32(publicKeySize), ss)
+	}
+	if err != nil {
+		panic(err)
+	}
+	_, err = h.Write(pubkey1)
+	if err != nil {
+		panic(err)
+	}
+	_, err = h.Write(pubkey2)
+	if err != nil {
+		panic(err)
+	}
+	ss2 := make([]byte, len(ss))
+	_, err = h.Read(ss2)
+	if err != nil {
+		panic(err)
+	}
+	return ss2
+}
+
+// Returns the shared key encapsulated in ciphertext ct for the
+// private key sk.
+// Implements DECAPSULATE as described in NIKE to KEM adapter,
+// see docs/specs/kemsphinx.rst
+func (a *Scheme) Decapsulate(myPrivkey kem.PrivateKey, ct []byte) ([]byte, error) {
+	if len(ct) != a.CiphertextSize() {
+		return nil, kem.ErrCiphertextSize
+	}
+	theirPubkey, err := a.UnmarshalBinaryPublicKey(ct)
+	if err != nil {
+		return nil, err
+	}
+	// s = DH(my_priv_key, their_pub_key)
+	ss := a.nike.DeriveSecret(myPrivkey.(*PrivateKey).privateKey, theirPubkey.(*PublicKey).publicKey)
+	// shared_key = H(ss || my_pub_key || their_pub_key)
+	ss2 := hashSharedSecretWithPublicKeys(a.nike.PublicKeySize(), ss, myPrivkey.Public().(*PublicKey).publicKey.Bytes(), theirPubkey.(*PublicKey).publicKey.Bytes())
+	return ss2, nil
+}
+
+// Unmarshals a PublicKey from the provided buffer.
+func (a *Scheme) UnmarshalBinaryPublicKey(b []byte) (kem.PublicKey, error) {
+	pubkey, err := a.nike.UnmarshalBinaryPublicKey(b)
+	if err != nil {
+		return nil, err
+	}
+	return &PublicKey{
+		publicKey: pubkey,
+		scheme:    a,
+	}, nil
+}
+
+// Unmarshals a PrivateKey from the provided buffer.
+func (a *Scheme) UnmarshalBinaryPrivateKey(b []byte) (kem.PrivateKey, error) {
+	privkey, err := a.nike.UnmarshalBinaryPrivateKey(b)
+	if err != nil {
+		return nil, err
+	}
+	return &PrivateKey{
+		privateKey: privkey,
+		scheme:     a,
+	}, nil
+}
+
+// Size of encapsulated keys.
+func (a *Scheme) CiphertextSize() int {
+	return a.nike.PublicKeySize()
+}
+
+// Size of established shared keys.
+func (a *Scheme) SharedKeySize() int {
+	return a.nike.PublicKeySize()
+}
+
+// Size of packed private keys.
+func (a *Scheme) PrivateKeySize() int {
+	return a.nike.PrivateKeySize()
+}
+
+// Size of packed public keys.
+func (a *Scheme) PublicKeySize() int {
+	return a.nike.PublicKeySize()
+}
+
+// DeriveKeyPair deterministicallly derives a pair of keys from a seed.
+// Panics if the length of seed is not equal to the value returned by
+// SeedSize.
+func (a *Scheme) DeriveKeyPair(seed []byte) (kem.PublicKey, kem.PrivateKey) {
+	if len(seed) != a.SeedSize() {
+		panic(fmt.Errorf("%s: provided seed of length %d is != to correct seed size of %d", kem.ErrSeedSize, len(seed), a.SeedSize()))
```

Comment:
```suggestion
		panic(fmt.Errorf("%s: provided len(seed) %d != a.SeedSize() %d", kem.ErrSeedSize, len(seed), a.SeedSize()))
```
Mentioning the parameter name and the constant it's supposed to match seems like an easy life quality improvement for whoever triggers this assert here :)

Reply:
fixed